src/main/java/me/qoomon/maven/gitversioning/GitVersioningModelProcessor.java

```diff
@@ -152,17 +152,6 @@ public Model processModel(Model projectModel, Map<String, ?> options) throws IOE
                 logger.debug("configFile: " + configFile.toString());
                 config = loadConfig(configFile);
 
-                gitDirectory = findGitDir(executionRootDirectory);
-                if (gitDirectory == null || !gitDirectory.exists()) {
-                    logger.warn("skip - project is not part of a git repository");
-                    disabled = true;
-                    return projectModel;
-                }
-
-                logger.debug("gitDirectory: " + gitDirectory.toString());
-
-                gitVersionDetails = getGitVersionDetails(config, executionRootDirectory);
```

Comment:
it is important to determine `gitVersionDetails` only once, otherwise it can cause version conflicts with different versions on different git repositories within on project. Currently the is no support for multiple versions in on maven multi-module project.

Reply:
But this is exactly what I need.

Your code is very clean, 
I only had to move `gitVersionDetails` from private class variable to a method variable, that's all. And It work's for my very good.

At the moment I have to provide a patched version from your extension. Unfortunately maven looks only in `central` for extensions, that's why either I have to fork and publish your extension or our contributors have to set a `mirror` for central in their local setting.

One option would be to provide this "submodule support" as an opt-in config setting.

Reply:
I probably will have some time at this weekend to do some testing.
I think I need to cache the generate version somewhere to be able to replace dependency versions with the right verion.

Reply:
Great news!
But you should provide this "submodule support" as an opt-in feature.